$OpenBSD$

strcat -> strlcat
ATTENTION: not sure if size for
sprintf -> snprintf is right

Index: codemp/rd-dedicated/tr_model.cpp
--- codemp/rd-dedicated/tr_model.cpp.orig
+++ codemp/rd-dedicated/tr_model.cpp
@@ -616,7 +616,7 @@ void RE_InsertModelIntoHash(const char *name, model_t 
 
 	mh->next = mhHashTable[hash];
 	mh->handle = mod->index;
-	strcpy(mh->name, name);
+	strlcpy(mh->name, name, sizeof(mh->name));
 	mhHashTable[hash] = mh;
 }
 /*
@@ -1016,7 +1016,7 @@ Ghoul2 Insert End
 	for ( lod = iLODStart; lod >= 0 ; lod-- ) {
 		char filename[1024];
 
-		strcpy( filename, name );
+		strlcpy( filename, name , sizeof(filename));
 
 		if ( lod != 0 ) {
 			char namebuf[80];
@@ -1024,8 +1024,8 @@ Ghoul2 Insert End
 			if ( strrchr( filename, '.' ) ) {
 				*strrchr( filename, '.' ) = 0;
 			}
-			sprintf( namebuf, "_%d.md3", lod );
-			strcat( filename, namebuf );
+			snprintf( namebuf, 256, "_%d.md3", lod );	// switch to snprintf, with size of 256
+			strlcat( filename, namebuf, sizeof(filename) );
 		}
 
 		qboolean bAlreadyCached = qfalse;
@@ -1227,7 +1227,7 @@ Ghoul2 Insert End
 	for ( lod = iLODStart; lod >= 0 ; lod-- ) {
 		char filename[1024];
 
-		strcpy( filename, name );
+		strlcpy( filename, name , sizeof(filename));
 
 		if ( lod != 0 ) {
 			char namebuf[80];
@@ -1236,7 +1236,7 @@ Ghoul2 Insert End
 				*strrchr( filename, '.' ) = 0;
 			}
 			sprintf( namebuf, "_%d.md3", lod );
-			strcat( filename, namebuf );
+			strlcat( filename, namebuf , sizeof(filename));
 		}
 
 		qboolean bAlreadyCached = qfalse;
